lib/services/ios-project-service.ts

```diff
@@ -221,11 +219,55 @@ class IOSProjectService implements  IPlatformProjectService {
 			shell.cp("-R", path.join(cachedPackagePath, "*"), path.join(this.platformData.projectRoot, util.format("%s.xcodeproj", this.$projectData.projectName)));
 			this.$logger.info("Copied from %s at %s.", cachedPackagePath, this.platformData.projectRoot);
 
-
 			var pbxprojFilePath = path.join(this.platformData.projectRoot, this.$projectData.projectName + IOSProjectService.XCODE_PROJECT_EXT_NAME, "project.pbxproj");
 			this.replaceFileContent(pbxprojFilePath).wait();
 		}).future<void>()();
 	}
+	
+	public prepareProject(): IFuture<void> {
+		return (() => {
+			let project = this.pbxProj;
+			let resources = project.pbxGroupByName("Resources");
+			
+			if(resources) {
+				let references = project.pbxFileReferenceSection();
+				
+				let xcodeProjectImages = _.map(<any[]>resources.children, resource => references[resource.value].name.replace(/\"/g, ""));
+				this.$logger.trace("Images from Xcode project");
+				this.$logger.trace(xcodeProjectImages);
+				
+				let appResourcesImages = this.$fs.readDirectory(this.platformData.appResourcesDestinationDirectoryPath).wait();
+				this.$logger.trace("Current images from App_Resources");
+				this.$logger.trace(appResourcesImages);
+	
+				let imagesToAdd = _.difference(appResourcesImages, xcodeProjectImages);
+				this.$logger.trace(`New images to add into xcode project: ${imagesToAdd.join(", ")}`);			
+				_.each(imagesToAdd, image => project.addResourceFile(path.join(this.platformData.appResourcesDestinationDirectoryPath, image)));
+				
+				let imagesToRemove = _.difference(xcodeProjectImages, appResourcesImages);
+				this.$logger.trace(`Images to remove from xcode project: ${imagesToRemove.join(", ")}`);			
+				_.each(imagesToRemove, image => project.removeResourceFile(path.join(this.platformData.appResourcesDestinationDirectoryPath, image))); 
+				
+				this.savePbxProj(project).wait();
+			}
+			
+		}).future<void>()();
+	}
+	
+	private get pbxProjPath(): string {
+		return path.join(this.platformData.projectRoot, this.$projectData.projectName + ".xcodeproj", "project.pbxproj");
+	}
+	
+	private get pbxProj(): any {
+		let project = new xcode.project(this.pbxProjPath);
```

Comment:
This function is called only one time - https://github.com/NativeScript/nativescript-cli/pull/590/files#diff-b673f137deb4afccccc0bd5f1572c2a0R229